torch_np/_funcs_impl.py

```diff
@@ -36,6 +38,13 @@ def copy(a: ArrayLike, order="K", subok: SubokLike = False):
     return a.clone()
 
 
+def copyto(dst: NDArray, src: ArrayLike, casting="same_kind", where=NoValue):
+    if where is not NoValue:
+        raise NotImplementedError
+    (src,) = _util.typecast_tensors((src,), dst.tensor.dtype, casting=casting)
+    dst.tensor.copy_(src)
```

Comment:
Actually, this should be `dst.copy_(src)`?

Reply:
Not really. `dst` is annotated as NDArray and those do not decay to torch.Tensors. src is an ArrayLike, so it can be a list to be normalized to a tensor. But `dst` cannot be a generic ArrayLike and must be an ndarray.

Reply:
Sure, it must be an ndarray, but at this point, after the normalizer, we should have just PyTorch-land objects. In particular, we should never get `tnp.ndarray`s of any kind, just Tensors.

Reply:
`dst: NDArray` annotation only checks that its argument is an ndarray exactly and does not convert it to a Tensor:

https://github.com/Quansight-Labs/numpy_pytorch_interop/blob/main/torch_np/_normalizations.py#L62

Reply:
Yeah. My point then is that `NDArray` should convert it to a `Tensor`. Why would it not do so?

Reply:
Why would it. It's a rare annotation for when need to only accept an ndarray not a general array_like. And then we know it has a `.tensor` attribute. It's a rare annotation, currently used here and for the out= arrays, which also need to be exactly arrays.

Reply:
It currently breaks the invariant that every input to these functions is something that lives in PyTorch-land. I would expect this to be a more constrained version of `ArrayLike`, but currently it has other semantics, which is a bit odd.

Reply:
Okay. So does it hold gh-105?

Reply:
It can be done afterwards as a small ENH.

Reply:
https://github.com/Quansight-Labs/numpy_pytorch_interop/pull/108